src/source/vector_tile_source.js

```diff
@@ -94,6 +94,24 @@ class VectorTileSource extends Evented implements Source {
         this.load();
     }
 
+    setSourceProperty(name: string, value: mixed) {
+        if (this._tileJSONRequest) {
+            this._tileJSONRequest.cancel();
+        }
+
+        const options = { [name]: value};
+        extend(this, pick(options, ['url', 'scheme', 'tileSize']));
```

Comment:
I think it would be nicer if we could avoid the duplication of code between here and the constructor.

Reply:
I've moved options setter into `_updateOptions` method